### PR TITLE
Various multiple persistence units adjustments

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

**Comment by gsmet** [comment]
Failure on Windows is unrelated.

File: docs/src/main/asciidoc/hibernate-orm.adoc [hunk 1]
```diff
@@ -822,9 +822,36 @@ public class CustomTenantResolver implements TenantResolver {
...
+@PersistenceUnit("name-of-persistence-unit") <1>
+@RequestScoped
+@Unremovable
+public class CustomTenantResolver implements TenantResolver {
+
+   // ...
+
+}
+----
+<1> Be careful to use the `@io.quarkus.hibernate.orm.PersistenceUnit` annotation, not the JPA one.
```
Comment (geoand) [inline docs/src/main/asciidoc/hibernate-orm.adoc:852]: Minor suggestion: `Make sure to use the `@io.quarkus.hibernate.orm.PersistenceUnit` ...

File: integration-tests/hibernate-tenancy/src/main/java/io/quarkus/it/hibernate/multitenancy/inventory/InventoryTenantResolver.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.hibernate.orm.PersistenceUnit;
+import io.quarkus.hibernate.orm.runtime.tenant.TenantResolver;
+import io.vertx.ext.web.RoutingContext;
+
+@RequestScoped
+@Unremovable
```
Comment (geoand) [inline integration-tests/hibernate-tenancy/src/main/java/io/quarkus/it/hibernate/multitenancy/inventory/InventoryTenantResolver.java:14]: Is this needed? If so, then we probably want to mention it in the example in the docs as well

**Review by geoand — APPROVED** [review]
Just added some minor suggestions

File: docs/src/main/asciidoc/hibernate-orm.adoc (hunk 1 above)
Comment (gsmet) [reply docs/src/main/asciidoc/hibernate-orm.adoc]: Thanks, fixed.

File: integration-tests/hibernate-tenancy/src/main/java/io/quarkus/it/hibernate/multitenancy/inventory/InventoryTenantResolver.java (hunk 2 above)
Comment (gsmet) [reply integration-tests/hibernate-tenancy/src/main/java/io/quarkus/it/hibernate/multitenancy/inventory/InventoryTenantResolver.java]: So, it used to be like that and I didn't change it. I added a commit to make the beans automatically unremovable. Could you have a look at this commit?

Comment (geoand) [reply integration-tests/hibernate-tenancy/src/main/java/io/quarkus/it/hibernate/multitenancy/inventory/InventoryTenantResolver.java]: Seems perfectly reasonable!

**Pull request merged**